### PR TITLE
Add health checks for grafana add-on

### PR DESCRIPTION
**Pull request by Pothulapati**
Fourth part of #3979, as per #3979 (comment) Feel free to look at it, to understand how these changes integrate.

This PR adds health checks for grafana add-on, that are transient.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

**Comment by Pothulapati** [comment]
This PR updates the usual linkerd checks to also check for `linkerd-config-addons`, and based on the values from it `hc.addOns` can be used to perform add-on specific checks.

When `linkerd-config-addons` cm is not present, the checks would not fail but throw a warning as
```

linkerd-addons
--------------
‼ linkerd-config-addons config map exists
    configmaps "linkerd-config-addons" not found

Status check results are √

```

When the configmap is represent, as the grafana is enabled and its values are available, This PR adds grafana specific checks as following:

```
linkerd-addons
--------------
√ linkerd-config-addons config map exists

linkerd-grafana
---------------
√ grafana add-on service account exists
√ grafana add-on config map exists
√ grafana pod is running

Status check results are √
```

File: pkg/healthcheck/healthcheck_addons.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+					check: func(context.Context) error {
+						if grafana, ok := hc.addOns[l5dcharts.GrafanaAddOn]; ok {
+
+							// Get grafana add-on deployment
+							deployment, err := hc.kubeAPI.AppsV1().Deployments(hc.ControlPlaneNamespace).Get(grafana.(map[string]interface{})["name"].(string), metav1.GetOptions{})
+							if err != nil {
+								return err
+							}
+							// Get grafana pod with match labels
+							grafanaPods, err := hc.kubeAPI.CoreV1().Pods(hc.ControlPlaneNamespace).List(metav1.ListOptions{LabelSelector: labels.SelectorFromSet(deployment.Spec.Template.Labels).String()})
```
Comment (alpeb) [inline pkg/healthcheck/healthcheck_addons.go:82]: Why not leveraging `hc.controlPlanePods` instead of hitting the k8s API again?

Comment (Pothulapati) [reply pkg/healthcheck/healthcheck_addons.go]: Yep updated.

**Comment by Pothulapati** [comment]
@alpeb I am currently focusing to get these tests done for grafana and prometheus, but I could add similar tests for jaeger too! WDYT?

**Comment by alpeb** [comment]
Actually, I forgot, there's still the part to enable retries for this, while just issuing warnings if the addon didn't come up, correct?

**Comment by alpeb** [comment]
Thanks @Pothulapati, last changes look good to me :1st_place_medal: 
Let's get this merged! :trophy:

**Pull request merged**